packages/firestore/src/core/firestore_client.ts

```diff
@@ -234,6 +246,9 @@ export async function setOnlineComponentProvider(
   client.setCredentialChangeListener(user =>
     remoteStoreHandleCredentialChange(onlineComponentProvider.remoteStore, user)
   );
+  client.setAppCheckTokenChangeListener((appCheckToken, user) =>
```

Comment:
```suggestion
  client.setAppCheckTokenChangeListener((_, user) =>
```

Reply:
Does the AppCheckTokenListener really return the user?

Reply:
AppCheckTokenListener only provides the new app check token, not the user. I take the new app check token, as well as the latest user from the firestore_client to call `remoteStoreHandleCredentialChange`.

`remoteStoreHandleCredentialChange` needs to know the latest user because if the user has changed, it'll do a bunch of things related to user changes. By passing it the current user, that logic will be bypassed and only the restarting of the streams occurs.

Reply:
Hm, this is a bit of a strange API as the user has nothing to do with the AppCheck token. But I see that this simplifies things a bit, as you otherwise have to manage the current user in yet another place. Let's leave as is for now and remove once the backend fixes the underlying issue.